### PR TITLE
[MOBDW-82] ESLint fixes, new i18n for Menu

### PR DESCRIPTION
**Pull request by fktt22**
Soundtrack of this PR: [Saturn, "The Planets"](https://www.youtube.com/watch?v=Isic2Z2e2xs&t=1571s) This PR needs reflection (and inspiration!)

### Motivation

Menu needs i18n.
[MOBDW-82]

### In this PR

i18n was separated for Menu (main) and App (renderer). Special care was needed because i18next works as a singleton, so new, separate instances were needed.

Language was renamed to `en-US` to follow standard.

Minor fixes in several files were also applied; for instance, the style for callback declarations in the menu was unified.

### Caveats

For some reason, the menu changes do not (always) work -- at some time they did produce the right translations, but not now. It seems everything is loaded correctly, but translations still fail.

### Future Work

Find problem with main translations.

### Screen Shots

None needed.

### Testing

All tests pass.

[MOBDW-82]: https://mobilecoin.atlassian.net/browse/MOBDW-82

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ module.exports = {
         'ts-ignore': 'allow-with-description',
       },
     ],
-    'arrow-body-style': ['error', 'always'],
+    curly: ['error', 'all'],
```
Comment (fktt22) [inline .eslintrc.js:35]: This was done to match AirBNB standard, and to allow for shorter code in many callbacks.

File: app/app.html [hunk 2]
```diff
@@ -22,10 +23,7 @@
     <title>MobileCoin Desktop Wallet</title>
     <script>
       (() => {
-        if (
-          typeof process !== 'object' ||
-          (typeof process === 'object' && !process.env.START_HOT)
-        ) {
+        if (typeof process !== 'object' || !process.env.START_HOT) {
```
Comment (fktt22) [inline app/app.html:26]: This applied boolean equivalences: `X OR (NOT X AND Y)` equals `X OR Y`.

File: app/i18n.ts [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
 
-i18n
-  // pass the i18n instance to react-i18next.
+newInstance // pass the i18n instance to react-i18next.
   .use(initReactI18next)
-  // init i18next
   .init({
     debug: true,
-    fallbackLng: 'enUS',
+    fallbackLng: 'en-US',
```
Comment (fktt22) [inline app/i18n.ts:12]: For i18n, languages should always be specified as `nn-NN` -- language code first, in lowercase, then a hyphen, and finally country code second, in uppercase.

File: app/index.tsx [hunk 4]
```diff
@@ -3,15 +3,14 @@ import React, { Fragment } from 'react';
 import { render } from 'react-dom';
 import { AppContainer as ReactHotAppContainer } from 'react-hot-loader';
 
+import App from './App';
```
Comment (fktt22) [inline app/index.tsx:6]: To avoid using `require()` and go with the standard `import` style

File: locales/en-US.json [hunk 5]
```diff
@@ -11,10 +11,12 @@
     "mobUrlTooltip": "Show MobURL QR Code",
     "unnamed": "Unnamed Account"
   },
+
```
Comment (fktt22) [inline locales/en-US.json:14]: This empty lines are for clarity, to make reading easier.

File: app/i18n-node.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import i18n from 'i18n';
```
Comment (fktt22) [inline app/i18n-node.ts:1]: This is the standard `i18n` for node -- with this, everything in the menu works OK. I still don't know why i18next doesn't produce translations, so I went for using `i18n` instead for the menu.

File: app/i18n-main.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import i18next from 'i18next';
```
Comment (davidyhall) [inline app/i18n-main.ts:1]: Just a reminder to drop this file before merging.

File: app/i18n-node.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  defaultLocale: 'en',
+  fallbacks: { 'en-*': 'en' },
+  objectNotation: true,
+  staticCatalog: {
+    en: enUS,
+    'en-US': enUS,
+  },
+});
+
+/* TO DO: remove this when everything's fine:
```
Comment (davidyhall) [inline app/i18n-node.ts:15]: Reminder to delete.

File: app/i18n.ts (hunk 3 above)
Comment (davidyhall) [reply app/i18n.ts:12]: SGTM!

File: app/menu.ts [hunk 9]
```diff
@@ -1,5 +1,15 @@
 import { app, Menu, shell, BrowserWindow, MenuItemConstructorOptions } from 'electron';
 
+import i18n from './i18n-node';
+
+// eslint-disable-next-line no-underscore-dangle
+const t = (x: string) => i18n.__(`Menu.${x}`);
+
+/* TO DO: remove this when everything's final
```
Comment (davidyhall) [inline app/menu.ts:8]: Reminder to remove

File: test/components/AuthFlowGuard.test.tsx [hunk 10]
```diff
@@ -8,15 +8,15 @@ import renderSnapshot from '../renderSnapshot';
 
 jest.mock('../../app/hooks/useMobileCoinD');
 
-function setupComponent(contextOverides?: MobileCoinDContextValue) {
+function setupComponent(contextOverrides?: MobileCoinDContextValue) {
```
Comment (davidyhall) [inline test/components/AuthFlowGuard.test.tsx:11]: 💯  -- I'll do a lot of typos.

**Review by davidyhall — COMMENTED** [review]
I'm getting this error when I try to `yarn package`:

```
Uncaught Exception:
TypeError: he.configure is not a function
    at Module../app/main.dev.ts (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:44902)
    at r (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:124)
    at module.exports../app/main.dev.ts (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:923)
    at Object.<anonymous> (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:950)
    at Module._compile (internal/modules/cjs/loader.js:1152:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1173:10)
    at Module.load (internal/modules/cjs/loader.js:992:32)
    at Module._load (internal/modules/cjs/loader.js:885:14)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Object.<anonymous> (electron/js2c/browser_init.js:185:3510)
``` 

And this when running `yarn build`:

App threw an error during load
TypeError: he.configure is not a function
    at Module../app/main.dev.ts /desktop-wallet/app/main.prod.js:2:44902)
    at r (/desktop-wallet/app/main.prod.js:2:124)
    at module.exports../app/main.dev.ts (/desktop-wallet/app/main.prod.js:2:923)
    at Object.<anonymous> (/desktop-wallet/app/main.prod.js:2:950)
    at Module._compile (internal/modules/cjs/loader.js:1152:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1173:10)
    at Module.load (internal/modules/cjs/loader.js:992:32)
    at Module._load (internal/modules/cjs/loader.js:885:14)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at loadApplicationPackage (/desktop-wallet/node_modules/electron/dist/Electron.app/Contents/Resources/default_app.asar/main.js:110:1
```

Any idea?

File: app/i18n-main.ts (hunk 7 above)
Comment (fktt22) [reply app/i18n-main.ts:1]: I didn't want to drop `i18-main.ts` itself just in case a solution was found... but I'll do that anyway.

**Comment by fktt22** [comment]
> I'm getting this error when I try to `yarn package`:
> 
> ```
> Uncaught Exception:
> TypeError: he.configure is not a function
>     at Module../app/main.dev.ts (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:44902)
>     at r (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:124)
>     at module.exports../app/main.dev.ts (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:923)
>     at Object.<anonymous> (/Applications/MobileCoin Desktop Wallet beta-0.2.0.app/Contents/Resources/app.asar/main.prod.js:2:950)
>     at Module._compile (internal/modules/cjs/loader.js:1152:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1173:10)
>     at Module.load (internal/modules/cjs/loader.js:992:32)
>     at Module._load (internal/modules/cjs/loader.js:885:14)
>     at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
>     at Object.<anonymous> (electron/js2c/browser_init.js:185:3510)
> ```
> 
> And this when running `yarn build`:
> 
> ```
> App threw an error during load
> TypeError: he.configure is not a function
>     at Module../app/main.dev.ts /desktop-wallet/app/main.prod.js:2:44902)
>     at r (/desktop-wallet/app/main.prod.js:2:124)
>     at module.exports../app/main.dev.ts (/desktop-wallet/app/main.prod.js:2:923)
>     at Object.<anonymous> (/desktop-wallet/app/main.prod.js:2:950)
>     at Module._compile (internal/modules/cjs/loader.js:1152:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1173:10)
>     at Module.load (internal/modules/cjs/loader.js:992:32)
>     at Module._load (internal/modules/cjs/loader.js:885:14)
>     at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
>     at loadApplicationPackage (/desktop-wallet/node_modules/electron/dist/Electron.app/Contents/Resources/default_app.asar/main.js:110:16
> ```
> 
> Any idea?

Tried `yarn build` -- works fine (at least now) 

```

> [0] yarn build-main exited with code 0
> [1] Hash: 0e825fae2b71637cf3fb
> [1] Version: webpack 4.46.0
> [1] Time: 7854ms
> [1] Built at: 02/16/2021 8:13:49 PM
> [1]                                  Asset      Size  Chunks                         Chunk Names
> [1]   0454203f26b33fc02e2b686b317aab3d.svg   896 KiB          [emitted] [immutable]  
> [1] 2c154b0f8c0d8d1661627d1ddb317b12.woff2  13.3 KiB          [emitted] [immutable]  
> [1]   330e879afe4a0abb35f235e29be3084f.ttf   133 KiB          [emitted] [immutable]  
> [1] 3eb06c702e27fb110194f5a16c45cb8e.woff2  78.4 KiB          [emitted] [immutable]  
> [1]  5f63cb7f47b6ea89773b43a6e687e5a5.woff    90 KiB          [emitted] [immutable]  
> [1]   62a07ffeac77696f17ef438f49ce6790.eot  33.5 KiB          [emitted] [immutable]  
> [1]   6606667d9800a27eb8b5f61ccb66d510.eot   200 KiB          [emitted] [immutable]  
> [1] 6e63bd22128f27b83f228bf5ef541156.woff2  76.6 KiB          [emitted] [immutable]  
> [1]   915a0b79c22a1c1f64da9e0a90a12f02.ttf   200 KiB          [emitted] [immutable]  
> [1]   98f20b9ec79b2fee02a300f4b716629f.eot   134 KiB          [emitted] [immutable]  
> [1]   991c1c761fc31f9c3252dbfb2a22fd7a.svg   730 KiB          [emitted] [immutable]  
> [1]   ac2367644e559de4ff330fbb7c273e70.ttf  33.3 KiB          [emitted] [immutable]  
> [1]  ea5a41ec4a24ce93298ee053b6357e18.woff  16.4 KiB          [emitted] [immutable]  
> [1]   f3187c7462849ed261a89dc27e7a4733.svg   141 KiB          [emitted] [immutable]  
> [1]  f4f93856730733912b1e06ad64c0baf7.woff   102 KiB          [emitted] [immutable]  
> [1]                       renderer.prod.js  2.47 MiB       0  [emitted]              main
> [1]           renderer.prod.js.LICENSE.txt  2.92 KiB          [emitted]              
> [1]                              style.css  58.3 KiB       0  [emitted]              main
> [1]                          style.css.map   109 KiB          [emitted]              
> [1] Entrypoint main = style.css renderer.prod.js
> [1] [0] multi core-js regenerator-runtime/runtime ./app/index.tsx 52 bytes {0} [built]
> [1] [./app/index.tsx] ./app/index.tsx + 500 modules 1.9 MiB {0} [built]
> [1]     | ./app/index.tsx 1.4 KiB [built]
> [1]     | ./app/App.tsx 1.92 KiB [built]
> [1]     | ./app/app.global.css 50 bytes [built]
> [1]     | ./app/i18n.ts 539 bytes [built]
> [1]     | ./app/constants/themes.ts 534 bytes [built]
> [1]     | ./app/contexts/MobileCoinDContext.tsx 39.5 KiB [built]
> [1]     | ./app/mobilecoind/client.ts 1.33 KiB [built]
> [1]     | ./app/routes.tsx 4.45 KiB [built]
> [1]     | ./app/theme/index.ts 2.13 KiB [built]
> [1]     | ./app/constants/colors.ts 426 bytes [built]
> [1]     | ./app/mobilecoind/api/index.ts 1 KiB [built]
> [1]     | ./app/mobilecoind/services/BuildGiftCodeService.ts 7.14 KiB [built]
> [1]     | ./app/mobilecoind/services/BuildTransactionService.ts 8.89 KiB [built]
> [1]     | ./app/utils/LocalStore.ts 4.49 KiB [built]
> [1]     | ./app/mobilecoind/services/DecryptEntropyService.ts 6.57 KiB [built]
> [1]     |     + 486 hidden modules
> [1] [./app/mobilecoind/protos/blockchain_pb.js] 52.3 KiB {0} [built]
> [1] [./app/mobilecoind/protos/external_pb.js] 177 KiB {0} [built]
> [1] [./app/mobilecoind/protos/mobilecoind_api_grpc_pb.js] 42.8 KiB {0} [built]
> [1] [./app/mobilecoind/protos/mobilecoind_api_pb.js] 451 KiB {0} [built]
> [1] [./locales/en-US.json] 18 KiB {0} [built]
> [1] [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
> [1] [assert] external "assert" 42 bytes {0} [built]
> [1] [crypto] external "crypto" 42 bytes {0} [built]
> [1] [electron] external "electron" 42 bytes {0} [built]
> [1] [http2] external "http2" 42 bytes {0} [built]
> [1] [path] external "path" 42 bytes {0} [built]
> [1] [tls] external "tls" 42 bytes {0} [built]
> [1] [util] external "util" 42 bytes {0} [built]
> [1]     + 1312 hidden modules
> [1] Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!app/app.global.css:
> [1]                                      Asset      Size  Chunks                         Chunk Names
> [1]       0454203f26b33fc02e2b686b317aab3d.svg   896 KiB          [emitted] [immutable]  
> [1]     2c154b0f8c0d8d1661627d1ddb317b12.woff2  13.3 KiB          [emitted] [immutable]  
> [1]       330e879afe4a0abb35f235e29be3084f.ttf   133 KiB          [emitted] [immutable]  
> [1]     3eb06c702e27fb110194f5a16c45cb8e.woff2  78.4 KiB          [emitted] [immutable]  
> [1]      5f63cb7f47b6ea89773b43a6e687e5a5.woff    90 KiB          [emitted] [immutable]  
> [1]       62a07ffeac77696f17ef438f49ce6790.eot  33.5 KiB          [emitted] [immutable]  
> [1]       6606667d9800a27eb8b5f61ccb66d510.eot   200 KiB          [emitted] [immutable]  
> [1]     6e63bd22128f27b83f228bf5ef541156.woff2  76.6 KiB          [emitted] [immutable]  
> [1]       915a0b79c22a1c1f64da9e0a90a12f02.ttf   200 KiB          [emitted] [immutable]  
> [1]       98f20b9ec79b2fee02a300f4b716629f.eot   134 KiB          [emitted] [immutable]  
> [1]       991c1c761fc31f9c3252dbfb2a22fd7a.svg   730 KiB          [emitted] [immutable]  
> [1]       ac2367644e559de4ff330fbb7c273e70.ttf  33.3 KiB          [emitted] [immutable]  
> [1]      ea5a41ec4a24ce93298ee053b6357e18.woff  16.4 KiB          [emitted] [immutable]  
> [1]       f3187c7462849ed261a89dc27e7a4733.svg   141 KiB          [emitted] [immutable]  
> [1]      f4f93856730733912b1e06ad64c0baf7.woff   102 KiB          [emitted] [immutable]  
> [1]     Entrypoint mini-css-extract-plugin = *
> [1]     [./node_modules/css-loader/dist/cjs.js?!./app/app.global.css] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./app/app.global.css 1.6 KiB {0} [built]
> [1]         + 19 hidden modules
> [1] yarn build-renderer exited with code 0
> Done in 9.38s.

```

File: app/menu.ts (hunk 9 above)
Comment (fktt22) [reply app/menu.ts]: Done

File: app/i18n-node.ts (hunk 8 above)
Comment (fktt22) [reply app/i18n-node.ts:15]: Done

File: app/i18n-main.ts (hunk 7 above)
Comment (fktt22) [reply app/i18n-main.ts]: Done

**Comment by davidyhall** [comment]
I've confirmed the issue on other macOS. I want to loop back on this PR in the next week.

**Comment by BrianCorbin** [comment]
I would suggest closing this branch, as this is breaking on Mac. Issue has been resolved and (most of) the changes implemented here are replicated in [PR#49](https://github.com/mobilecoinofficial/desktop-wallet/pull/49)

**Pull request closed without merging**